beta/src/components/MDX/Sandpack/DownloadButton.tsx

```diff
@@ -7,23 +7,33 @@ import {useSandpack} from '@codesandbox/sandpack-react';
 import {IconArrowSmall} from '../../Icon/IconArrowSmall';
 export interface DownloadButtonProps {}
 
-let initialIsSupported = false;
+let supportsImportMap;
 
-export const DownloadButton: React.FC<DownloadButtonProps> = () => {
-  const {sandpack} = useSandpack();
-  const [supported, setSupported] = React.useState(initialIsSupported);
-  React.useEffect(() => {
-    // This detection will work in Chrome 97+
-    if (
-      !supported &&
-      (HTMLScriptElement as any).supports &&
-      (HTMLScriptElement as any).supports('importmap')
-    ) {
-      setSupported(true);
-      initialIsSupported = true;
+function useSupportsImportMap() {
+  function subscribe() {
+    // It never updates.
+  }
+  function getCurrentValue() {
+    if (supportsImportMap === undefined) {
+      supportsImportMap =
+        (HTMLScriptElement as any).supports &&
+        (HTMLScriptElement as any).supports('importmap');
     }
-  }, [supported]);
+    return supportsImportMap;
+  }
+  function getServerSnapshot() {
+    return false;
+  }
+  return React.useSyncExternalStore(
```

Comment:
useSES seems like the way to go here since the value may change between sandboxes mounting. So it's not legit to cache it and then read the value that could be inconsistent from the server into useState.